multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ServiceBindingParametersGetter.java

```diff
@@ -0,0 +1,140 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.MessageFormat;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.CloudOperationException;
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.cloudfoundry.client.lib.domain.CloudEntity;
+import org.cloudfoundry.client.lib.domain.CloudServiceBinding;
+import org.cloudfoundry.client.lib.domain.CloudServiceInstance;
+import org.cloudfoundry.multiapps.common.SLException;
+import org.cloudfoundry.multiapps.common.util.JsonUtil;
+import org.cloudfoundry.multiapps.common.util.MapUtil;
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudApplicationExtended;
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudServiceInstanceExtended;
+import org.cloudfoundry.multiapps.controller.core.helpers.MtaArchiveElements;
+import org.cloudfoundry.multiapps.controller.core.security.serialization.SecureSerialization;
+import org.cloudfoundry.multiapps.controller.persistence.services.FileContentProcessor;
+import org.cloudfoundry.multiapps.controller.persistence.services.FileService;
+import org.cloudfoundry.multiapps.controller.persistence.services.FileStorageException;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.steps.ProcessContext;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.cloudfoundry.multiapps.mta.handlers.ArchiveHandler;
+import org.cloudfoundry.multiapps.mta.util.NameUtil;
+import org.springframework.http.HttpStatus;
+
+public class ServiceBindingParametersGetter {
+
+    private final ProcessContext context;
+    private final FileService fileService;
+    private final long maxManifestSize;
+
+    public ServiceBindingParametersGetter(ProcessContext context, FileService fileService, long maxManifestSize) {
+        this.context = context;
+        this.fileService = fileService;
+        this.maxManifestSize = maxManifestSize;
+    }
+
+    public Map<String, Object> getServiceBindingParametersFromMta(CloudApplicationExtended app, String serviceName)
+        throws FileStorageException {
+        Optional<CloudServiceInstanceExtended> optionalService = getService(context.getVariable(Variables.SERVICES_TO_BIND), serviceName);
```

Comment:
One idea is to change naming only to service and consequent check to !service.isPresent()
it is to not mislead this service to the optional services term